torch/nn/modules/sparse.py

```diff
@@ -188,9 +188,9 @@ class EmbeddingBag(Module):
 
     For bags of constant length and no :attr:`per_sample_weights`, this class
 
-        * with ``mode="sum"`` is equivalent to :class:`~torch.nn.Embedding` followed by ``torch.sum(dim=0)``,
-        * with ``mode="mean"`` is equivalent to :class:`~torch.nn.Embedding` followed by ``torch.mean(dim=0)``,
-        * with ``mode="max"`` is equivalent to :class:`~torch.nn.Embedding` followed by ``torch.max(dim=0)``.
+        * with ``mode="sum"`` is equivalent to :class:`~torch.nn.Embedding` followed by ``torch.sum(dim=1)``,
+        * with ``mode="mean"`` is equivalent to :class:`~torch.nn.Embedding` followed by ``torch.mean(dim=1)``,
+        * with ``mode="max"`` is equivalent to :class:`~torch.nn.Embedding` followed by ``torch.max(dim=1)``.
```

Comment:
I agree with your assessment that this isn't completely correct, because of how EmbeddingBag works with offsets.
1. without offsets: the input must be a 2D tensor and we're taking a sum on the last dimension
2. with offsets: the input must be a 1D tensor with offsets and we're taking a sum on every example in the batch.

We could do two things. The first is to clarify `For bags of constant length and no :attr:`per_sample_weights` and 2D inputs (with no offests)...`. The second is to say something like `mode=sum is equivalent to nn.Embedding followed by a sum of each example in the batch`

Reply:
If we choose to clarify `For bags of constant length and no :attr:per_sample_weights and 2D inputs (with no offsets) ...`, the reduction in that case is still on `dim=1`, right?
```
>>> import torch                                                                                     
>>> weights = torch.nn.Parameter(torch.randn(10, 3))
>>> e = torch.nn.Embedding(10, 3)                  
>>> eb = torch.nn.EmbeddingBag(10, 3, mode="sum")
>>> e.weight = weights                            
>>> eb.weight = weights                                                                              
>>> inputs = torch.LongTensor([[1, 7, 3, 4], [5, 0, 2, 6]])
>>> torch.sum(e(inputs), dim=1)
tensor([[ 0.1180,  0.9364, -0.5337],
        [ 0.7835, -2.5901,  3.0090]], grad_fn=<SumBackward1>)
>>> eb(inputs)
tensor([[ 0.1180,  0.9364, -0.5337],
        [ 0.7835, -2.5901,  3.0090]], grad_fn=<EmbeddingBagBackward>)    
```

Reply:
I added `and 2D inputs` but kept `dim=1`.